lib/term/src/Term/Maude/Signature.hs

```diff
@@ -116,9 +120,19 @@ instance Semigroup MaudeSig where
                            ,enableMSet=mset1||mset2
                            ,enableXor=xor1||xor2
                            ,enableDiff=diff1||diff2
-                           ,stFunSyms=S.union stFunSyms1 stFunSyms2
-                           ,stRules=S.union stRules1 stRules2})
-
+                           ,stFunSyms=unionExceptPairSym stFunSyms1 stFunSyms2
+                           ,stRules=unionExceptPairRules stRules1 stRules2})
+          -- an exception to merging is the destructor variants for pair, which is exclusive
+          -- in general, it might make sense to not merge fun syms with same identifier
+      where unionExceptPairSym st1 st2 = if pairFunDestSig `S.isSubsetOf` st2 then
```

Comment:
In general, without any change, maude signatures can contain duplicates of function symbols, for instance if you have two symbols:
```
totopub = ("totopub",(1, Public,Constructor))
totopriv = ("totopriv",(1, Private,Constructor))
```
Don't know what happens, and if the overall code actually allows to reach such weird cases.

For the specific case of pairing, we have to make sure that the basic pairing builtin is overwritten cleanly by the destructor version of it. And this would not happen without this particular change.